src/query/phrase_query/phrase_weight.rs

```diff
@@ -72,13 +73,53 @@ impl PhraseWeight {
                 }
             }
         }
-        Ok(Some(PhraseScorer::new(
+        Ok(Some(Box::new(ExactPhraseScorer::new(
             term_postings_list,
             similarity_weight,
             fieldnorm_reader,
             self.scoring_enabled,
             self.slop,
-        )))
+        ))))
+    }
+
+    fn phrase_scorer_as_scorer(
+        &self,
+        reader: &SegmentReader,
+        boost: Score,
+    ) -> crate::Result<Option<Box<dyn Scorer>>> {
```

Comment:
This is a copy/paste of `fn phrase_scorer`, juste the returned boxed dyn trait changes.